src/app/vertextool/qgsvertextool.cpp

```diff
@@ -1631,9 +1634,22 @@ void QgsVertexTool::applyEditsToLayers( QgsVertexTool::VertexEdits &edits )
     layer->beginEditCommand( tr( "Moved vertex" ) );
     QHash<QgsFeatureId, QgsGeometry>::iterator it2 = layerEdits.begin();
     for ( ; it2 != layerEdits.end(); ++it2 )
+    {
       layer->changeGeometry( it2.key(), it2.value() );
+      for ( int i = 0; i < mSelectedVertices.length(); ++i )
+      {
+        if ( mSelectedVertices.at( i ).layer == layer && mSelectedVertices.at( i ).fid == it2.value() )
+        {
+          mSelectedFeature->selectVertex( mSelectedVertices.at( i ).vertexId );
```

Comment:
just looking at this line, there is no chance that mSelectedFeature has previously other selected vertices? and that this selection needs to be cleaned first?

Reply:
AFAIK, no. And my test reseting the selectedfeature produces an other bug. So, I prefer don't touch it :)